docs/design/routing.md

```diff
@@ -0,0 +1,41 @@
+## Routing
+
+### Problem
+
+In order to expose MCP Servers, the MCP Gateway aggregates the available tools from the registered backend MCP servers and presents them as a single tool set to agent applications. As there is only one endpoint expected by the agent: `https://{host}/mcp` , the gateway needs to be able to accept MCP protocol based tools/call requests at that endpoint and then be able to make a routing decision about which backend MCP server should receive the call.
+
+
+### Solution
+
+> Note: In describing the solution [Gateway API](https://gateway-api.sigs.k8s.io/) and Kubernetes are used as the basis for defining and deploying routes and gateways. It is worth mentioning that the result of all these APIs is config that is eventually consumed by Envoy and as such could be done without the need for the Gateway API resources.
```

Comment:
Good to call this out.
Can work outside k8s & gwapi, but there's a nice integration with those when you have them to make things easier